src/Common/Commands.Common.Compute.Tests/CredentialManager.cs

```diff
@@ -0,0 +1,87 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Common.Compute.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.IdentityModel.Clients.ActiveDirectory;
+    using Microsoft.Rest;
+
+    class CredentialManager
+    {
+        protected CredentialManager() { }
+
+        private const string ServicePrincipalEnvVariableName = "AZURE_SERVICE_PRINCIPAL";
```

Comment:
We already have a test infrastructure that does this -this will make it essentially impossible to run these tests on ci without setting up this environment variable everywhere.  Let's dicuss thet best way to bring these tests inline.